src/components/views/dialogs/UserPickerDialog.js

```diff
@@ -15,48 +16,49 @@ limitations under the License.
 */
 
 import React from 'react';
+import PropTypes from 'prop-types';
 import { _t } from '../../../languageHandler';
 import sdk from '../../../index';
-import { getAddressType, inviteMultipleToRoom } from '../../../Invite';
-import createRoom from '../../../createRoom';
+import { getAddressType } from '../../../Invite';
 import MatrixClientPeg from '../../../MatrixClientPeg';
-import DMRoomMap from '../../../utils/DMRoomMap';
-import Modal from '../../../Modal';
 import AccessibleButton from '../elements/AccessibleButton';
 import Promise from 'bluebird';
-import dis from '../../../dispatcher';
+import { addressTypes } from '../../../Invite.js';
 
 const TRUNCATE_QUERY_LIST = 40;
 const QUERY_USER_DIRECTORY_DEBOUNCE_MS = 200;
 
 module.exports = React.createClass({
-    displayName: "ChatInviteDialog",
+    displayName: "UserPickerDialog",
+
     propTypes: {
-        title: React.PropTypes.string.isRequired,
-        description: React.PropTypes.oneOfType([
-            React.PropTypes.element,
-            React.PropTypes.string,
+        title: PropTypes.string.isRequired,
+        description: PropTypes.oneOfType([
+            PropTypes.element,
+            PropTypes.string,
         ]),
-        value: React.PropTypes.string,
-        placeholder: React.PropTypes.string,
-        roomId: React.PropTypes.string,
-        button: React.PropTypes.string,
-        focus: React.PropTypes.bool,
-        onFinished: React.PropTypes.func.isRequired,
+        value: PropTypes.string,
+        placeholder: PropTypes.string,
+        roomId: PropTypes.string,
+        button: PropTypes.string,
+        focus: PropTypes.bool,
+        validAddressTypes: PropTypes.arrayOf(PropTypes.oneOfType(addressTypes)),
+        onFinished: PropTypes.func.isRequired,
     },
 
     getDefaultProps: function() {
         return {
             value: "",
             focus: true,
+            validAddressTypes: addressTypes,
         };
     },
 
     getInitialState: function() {
         return {
             error: false,
 
-            // List of AddressTile.InviteAddressType objects representing
+            // List of InviteAddressType objects representing
```

Comment:
It's a bit odd having the word `invite` in variables, comments now that this component doesn't know about inviting